File: Easy/Brick in the Wall/Brick in the Wall.cs
using System;
using System.Collections.Generic;

class Solution
{
    static void Main(string[] args)
    {
        int X = int.Parse(Console.ReadLine());
        int N = int.Parse(Console.ReadLine());
        string[] inputs = Console.ReadLine().Split(' ');
        List<int> weights = new List<int>(N);
        double answer = 0;

        for (int i = 0; i < N; i++)  
        weights.Add(int.Parse(inputs[i]));
        
        weights.Sort((a, b) => b.CompareTo(a));

        for (int i = 0; i < weights.Count; i++) 
        answer += ((((i/X)+1)-1) * 6.5 / 100) * 10 * weights[i];

        Console.WriteLine(answer.ToString("0.000"));
    }
}

File: Easy/Brick in the Wall/Brick in the Wall.py
import sys
import math

x = int(input())
n = int(input())
weights = []
answer = 0
for i in input().split():
    weights.append(int(i))

weights.sort(reverse=True)

for i in range(0, len(weights)): 
        answer += (((int((i/x))+1)-1) * 6.5 / 100) * 10 * weights[i]

print(f'{answer:.3f}')